manifests/client.pp

```diff
@@ -216,6 +216,7 @@
   Hash $setenv_safe                           = {},
   String $up                                  = '',
   String $down                                = '',
+  String $environment                         = 'prod',
```

Comment:
Better not to use local class variables called $environment, as they might create confusion with Puppet's interla variable $environment  .
Use $env or something similar, here and in line 325.
Anyway, current code should work the same